CacheTagsBundle/Listener/CacheTagListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace lbarulski\CacheTagsBundle\Listener;
+
+use lbarulski\CacheTagsBundle\Annotation\CacheTag\AbstractCacheTag;
+use lbarulski\CacheTagsBundle\Annotation\CacheTag\RequestAttribute;
+use lbarulski\CacheTagsBundle\Service\Repository;
+use lbarulski\CacheTagsBundle\Tag\TagInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Doctrine\Common\Annotations\Reader;
+
+class CacheTagListener
+{
+	/** @var Reader */
+	protected $reader;
+
+	/** @var Repository */
+	private $repository;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Reader     $reader An Reader instance
+	 * @param Repository $repository
+	 */
+	public function __construct(Reader $reader, Repository $repository)
+	{
+		$this->reader     = $reader;
+		$this->repository = $repository;
+	}
+
+	/**
+	 * @param FilterResponseEvent $event
+	 */
+	public function onKernelController(FilterControllerEvent $event)
```

Comment:
please do separate `onKernelController` for handling plain tags and tags from request attributes, this will remove any `$configuration instanceof RequestAttribute` checking